`what4-{abc,blt}` fail to build due to `ConjMap`-related API changes

```
$ cabal build what4-abc what4-blt
Configuration is affected by the following files:
- cabal.project
- cabal.project.local
Build profile: -w ghc-9.4.8 -O1
In order, the following will be built (use -v for more details):
 - what4-abc-0.1 (lib) (first run)
 - what4-blt-0.2 (lib) (file src/What4/Solver/BLT.hs changed)
Preprocessing library for what4-blt-0.2...
Building library for what4-blt-0.2...
Preprocessing library for what4-abc-0.1...
Building library for what4-abc-0.1...
[1 of 1] Compiling What4.Solver.BLT ( src/What4/Solver/BLT.hs, /home/ryanscott/Documents/Hacking/Haskell/what4/dist-newstyle/build/x86_64-linux/ghc-9.4.8/what4-blt-0.2/build/What4/Solver/BLT.o, /home/ryanscott/Documents/Hacking/Haskell/what4/dist-newstyle/build/x86_64-linux/ghc-9.4.8/what4-blt-0.2/build/What4/Solver/BLT.dyn_o ) [Source file changed]
[1 of 1] Compiling What4.Solver.ABC ( src/What4/Solver/ABC.hs, /home/ryanscott/Documents/Hacking/Haskell/what4/dist-newstyle/build/x86_64-linux/ghc-9.4.8/what4-abc-0.1/build/What4/Solver/ABC.o, /home/ryanscott/Documents/Hacking/Haskell/what4/dist-newstyle/build/x86_64-linux/ghc-9.4.8/what4-abc-0.1/build/What4/Solver/ABC.dyn_o )

src/What4/Solver/BLT.hs:433:29: error:
    • Couldn't match expected type: BM.BoolMap (Expr t)
                  with actual type: BM.ConjMap (Expr t)
    • In the first argument of ‘BM.viewBoolMap’, namely ‘xs’
      In the expression: BM.viewBoolMap xs
      In the expression:
        case BM.viewBoolMap xs of
          BM.BoolMapUnit -> return ()
          BM.BoolMapDualUnit
            -> do when (isVerb h) $ warnAt l "problem assumes False"
                  setUNSAT h
          BM.BoolMapTerms (t :| ts)
            -> mapM_ f (t : ts)
            where
                f (x, Positive) = assume h x
                f (_, Negative) = unsupported
    • Relevant bindings include
        xs :: BM.ConjMap (Expr t) (bound at src/What4/Solver/BLT.hs:432:16)
        a :: App (Expr t) BaseBoolType
          (bound at src/What4/Solver/BLT.hs:430:7)
        ba :: AppExpr t BaseBoolType
          (bound at src/What4/Solver/BLT.hs:429:21)
        b :: BoolExpr t (bound at src/What4/Solver/BLT.hs:429:10)
        h :: Handle t (bound at src/What4/Solver/BLT.hs:429:8)
        assume :: Handle t -> BoolExpr t -> IO ()
          (bound at src/What4/Solver/BLT.hs:409:1)
    |
433 |         case BM.viewBoolMap xs of
    |                             ^^

src/What4/Solver/ABC.hs:373:27: error:
    • Couldn't match expected type: BM.BoolMap (Expr t)
                  with actual type: BM.ConjMap (Expr t)
    • In the first argument of ‘BM.viewBoolMap’, namely ‘xs’
      In the expression: BM.viewBoolMap xs
      In the expression:
        case BM.viewBoolMap xs of
          BM.BoolMapUnit -> return (B GIA.true)
          BM.BoolMapDualUnit -> return (B GIA.false)
          BM.BoolMapTerms (t :| ts)
            -> B <$> join (foldM (AIG.lAnd' g) <$> pol t <*> mapM pol ts)
    • Relevant bindings include
        pol :: (Expr t BaseBoolType, Polarity)
               -> IO (LitValue s BaseBoolType)
          (bound at src/What4/Solver/ABC.hs:370:11)
        xs :: BM.ConjMap (Expr t) (bound at src/What4/Solver/ABC.hs:369:14)
        ae :: AppExpr t tp (bound at src/What4/Solver/ABC.hs:278:16)
        h :: Network t s (bound at src/What4/Solver/ABC.hs:278:14)
        bitblastExpr :: Network t s -> AppExpr t tp -> IO (NameType s tp)
          (bound at src/What4/Solver/ABC.hs:278:1)
    |
373 |       case BM.viewBoolMap xs of
    |                           ^^
Error: [Cabal-7125]
Failed to build what4-abc-0.1.
Failed to build what4-blt-0.2.
```

--- a/what4-abc/src/What4/Solver/ABC.hs
+++ b/what4-abc/src/What4/Solver/ABC.hs
@@ -370,10 +370,10 @@ bitblastExpr h ae = do
       let pol (x,BM.Positive) = eval' h x
           pol (x,BM.Negative) = AIG.not <$> eval' h x
       in
-      case BM.viewBoolMap xs of
-        BM.BoolMapUnit -> return (B GIA.true)
-        BM.BoolMapDualUnit -> return (B GIA.false)
-        BM.BoolMapTerms (t:|ts) ->
+      case BM.viewConjMap xs of
+        BM.ConjTrue -> return (B GIA.true)
+        BM.ConjFalse -> return (B GIA.false)
+        BM.Conjuncts (t:|ts) ->
           B <$> join (foldM (AIG.lAnd' g) <$> pol t <*> mapM pol ts)
 
     SemiRingSum s ->

--- a/what4-blt/src/What4/Solver/BLT.hs
+++ b/what4-blt/src/What4/Solver/BLT.hs
@@ -430,12 +430,12 @@ assume h b@(AppExpr ba) =
   let a = appExprApp ba in
     case a of
       ConjPred xs ->
-        case BM.viewBoolMap xs of
-          BM.BoolMapUnit -> return ()
-          BM.BoolMapDualUnit ->
+        case BM.viewConjMap xs of
+          BM.ConjTrue -> return ()
+          BM.ConjFalse ->
                do when (isVerb h) $ warnAt l "problem assumes False"
                   setUNSAT h
-          BM.BoolMapTerms (t:|ts) -> mapM_ f (t:ts)
+          BM.Conjuncts (t:|ts) -> mapM_ f (t:ts)
             where f (x,BM.Positive) = assume h x
                   f (_,BM.Negative) = unsupported
 

